Draft/Classes/Draft.swift

```diff
@@ -36,14 +40,18 @@ public extension Draft {
     var session: URLSession { return .shared }
     
     func run() -> Request<ResponseType> {
-        guard var url = URL(string: host) else {
-            let request = Request<ResponseType>()
-            request.error = RequestError.badUrl(host)
-            return request
+        var components = URLComponents()
+        components.scheme = scheme
+        components.host = host
+        components.path = route
+        components.queryItems = parameters.map { (arg) -> URLQueryItem in
+            URLQueryItem(name: arg.key, value: arg.value.description)
         }
         
-        for component in route.split(separator: "/", omittingEmptySubsequences: true) {
-            url.appendPathComponent(String(component))
+        guard let url = components.url else {
+            let request = Request<ResponseType>()
+            request.error = RequestError.badUrl(components.string ?? host) // thoughts? better fail-over?
+            return request
         }
```

Comment:
Instead of this, ideally we combine our components into a string description ourselves (Draft should be CustomStringConvertible). It'd be nice if URLComponents would give us the malformed string anyway.
Feel free to merge it in as is, and we can fix this in a future PR.

Reply:
what about for description we do:` request to \(scheme)://\(host)\(route) with params \(parameters)`

Reply:
That looks good!